tests/controllers/buyerReview.test.ts

```diff
@@ -0,0 +1,141 @@
+      mockSellerService._id
+    );
+    expect(getBuyerReviewByIdQuery.data?.seller._id).toEqual(
+      mockSellerUser._id
+    );
+    expect(getBuyerReviewByIdQuery.data?.buyer._id).toEqual(mockBuyerUser._id);
+    expect(getBuyerReviewByIdQuery.data?.comment).toEqual(
+      mockBuyerReview.comment
+    );
+    expect(getBuyerReviewByIdQuery.data?.rating).toEqual(
+      mockBuyerReview.rating
+    );
+  });
+
+  it('Should be able to fetch BuyerReviews by Buyer User', async () => {
+    let getBuyerReviewByIdQuery = await getBuyerReviewsByBuyerUser(
+      mockBuyerUser._id.toString()
+    );
+    expect(getBuyerReviewByIdQuery.status).toBe(true);
+    expect(getBuyerReviewByIdQuery.data).toBeInstanceOf(Array);
+    expect(getBuyerReviewByIdQuery.data?.length).toBe(1);
+
+    getBuyerReviewByIdQuery = await getBuyerReviewsByBuyerUser(
+      mockSellerUser._id.toString()
+    );
+    expect(getBuyerReviewByIdQuery.status).toBe(true);
+    expect(getBuyerReviewByIdQuery.data).toBeInstanceOf(Array);
+    expect(getBuyerReviewByIdQuery.data?.length).toBe(0);
+  });
+
+  it('Should be able to fetch BuyerReviews by Seller User', async () => {
+    let getBuyerReviewByIdQuery = await getBuyerReviewsByPostingUser(
+      mockSellerUser._id.toString()
+    );
+    expect(getBuyerReviewByIdQuery.status).toBe(true);
+    expect(getBuyerReviewByIdQuery.data).toBeInstanceOf(Array);
+    expect(getBuyerReviewByIdQuery.data?.length).toBe(1);
+
+    getBuyerReviewByIdQuery = await getBuyerReviewsByPostingUser(
+      mockBuyerUser._id.toString()
+    );
+    expect(getBuyerReviewByIdQuery.status).toBe(true);
+    expect(getBuyerReviewByIdQuery.data).toBeInstanceOf(Array);
+    expect(getBuyerReviewByIdQuery.data?.length).toBe(0);
+  });
+
+  it('Should be able to create a BuyerReview', async () => {
+    let createBuyerReviewQuery = await createBuyerReview(
+      mockSellerUser._id.toString(),
+      mockBuyerRequest._id.toString(),
+      'Created inside the test',
+      3.6
+    );
+
+    expect(createBuyerReviewQuery.status).toBe(true);
+    expect(createBuyerReviewQuery.data).toBeInstanceOf(BuyerReview);
+    expect(createBuyerReviewQuery.data?.request).toEqual(mockBuyerRequest._id);
+    expect(createBuyerReviewQuery.data?.service).toEqual(mockSellerService._id);
+    expect(createBuyerReviewQuery.data?.seller).toEqual(mockSellerUser._id);
+    expect(createBuyerReviewQuery.data?.buyer).toEqual(mockBuyerUser._id);
+    expect(createBuyerReviewQuery.data?.comment).toEqual(
+      'Created inside the test'
+    );
+    expect(createBuyerReviewQuery.data?.rating).toEqual(3.6);
+  });
+
+  it('Should be able to delete a BuyerReview', async () => {
+    let deleteBuyerReviewQuery = await deleteBuyerReview(
+      mockBuyerReview._id.toString(),
+      mockSellerUser._id.toString()
+    );
+    expect(deleteBuyerReviewQuery.status).toBe(true);
+    expect(await BuyerReview.countDocuments({})).toEqual(1);
```

Comment:
You should check to see if the BuyerReview has been deleted. Even if the number of documents are 1, the BuyerReview need not have necessarily been deleted.

Reply:
Done